### ISSUE TITLE
MimeType/CharSet initialization fails on M1 architecture

### ISSUE THREAD

**Issue by Geooorg** [eclipse/jetty.project#6923]
This is a follow-up on https://github.com/localstack/localstack/issues/4625 
Localstack uses DynamoDB with Jetty.
The executable DynamoDB jar used by LocalStack is currently statically included from https://github.com/localstack/localstack-artifacts/blob/master/dynamodb-local-patch/etc/DynamoDBLocal.zip 

**Jetty version(s)**

This zip contains dependencies on Jetty 9.4.18.v20190429

**Java version/vendor** `(use: java -version)`

openjdk version "11.0.12" 2021-07-20
OpenJDK Runtime Environment (build 11.0.12+7-alpine-r0)
OpenJDK 64-Bit Server VM (build 11.0.12+7-alpine-r0, mixed mode)

**OS type/version**

Docker for OSX, 4.0.1 (68347)

**Description / How to reproduce?**

When Jetty is started from the DynamoDB, it fails to startup because the initialization process of MimeTypes and CharacterSets fails.

To reproduce, create a docker-compose file and run it on a MacOS with M1 architecture:

```version: '3'
services:

#  identity:
# Run the identity service inside of your local IDE.
# Make sure you're "swamped" in the correct AWS profile.

  redis:
    image: "redis:latest"
    ports:
      - "6379:6379"

#  dynamodb:
#    image: "amazon/dynamodb-local"
#    ports:
#      - "9050:8000"

  localstack:
    image: "localstack/localstack"
    ports:
      - "4566:4566"
    environment:
      SERVICES: "dynamodb"
      DEFAULT_REGION: "eu-central-1"
    healthcheck:
      test: [ "CMD", "curl", "--sS", "127.0.0.1:4566" ]
      interval: 20s
      timeout: 5s
      retries: 4```


Run 
```docker-compose up```

Or, inside the container:

```/opt/code/localstack/localstack/infra/dynamodb # java -Djava.library.path=./DynamoDBLocal_lib -Xmx256m -jar DynamoDBLocal.jar -port 57749 -inMemory```

This will fail giving the following stacktrace:

```Initializing DynamoDB Local with the following configuration:
Port:	57749
InMemory:	true
DbPath:	null
SharedDb:	false
shouldDelayTransientStatuses:	false
CorsParams:	*

Exception in thread "main" java.lang.ExceptionInInitializerError
	at org.eclipse.jetty.http.MimeTypes.<clinit>(MimeTypes.java:191)
	at org.eclipse.jetty.server.handler.ContextHandler.doStart(ContextHandler.java:836)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:167)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:119)
	at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:113)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:167)
	at org.eclipse.jetty.server.Server.start(Server.java:418)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:110)
	at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:113)
	at org.eclipse.jetty.server.Server.doStart(Server.java:382)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
	at com.amazonaws.services.dynamodbv2.local.server.DynamoDBProxyServer.start(DynamoDBProxyServer.java:83)
	at com.amazonaws.services.dynamodbv2.local.main.ServerRunner.main(ServerRunner.java:76)
Caused by: java.nio.charset.IllegalCharsetNameException: l;charset=iso-8859-1
	at java.base/java.nio.charset.Charset.checkName(Unknown Source)
	at java.base/java.nio.charset.Charset.lookup2(Unknown Source)
	at java.base/java.nio.charset.Charset.lookup(Unknown Source)
	at java.base/java.nio.charset.Charset.forName(Unknown Source)
	at org.eclipse.jetty.http.MimeTypes$Type.<init>(MimeTypes.java:113)
	at org.eclipse.jetty.http.MimeTypes$Type.<clinit>(MimeTypes.java:69)
	... 15 more
```

Striking is this line:
`Caused by: java.nio.charset.IllegalCharsetNameException: l;charset=iso-8859-1`
... as it looks like the substring in Jetty's MimeTypes.java doesn't work as expected. 

Some more thoughts are given in the parallel issue in Localstack (please see https://github.com/localstack/localstack/issues/4625).

```        Type(String s, Type base)
        {
            _string = s;
            _buffer = BufferUtil.toBuffer(s);
            _base = base;
            int i = s.indexOf(";charset=");
            _charset = Charset.forName(s.substring(i + 9));
            _charsetString = _charset.toString().toLowerCase(Locale.ENGLISH);
            _assumedCharset = false;
            _field = new PreEncodedHttpField(HttpHeader.CONTENT_TYPE, _string);
        }
```

I hope you're able to reproduce that and find a solution.

**Comment by joakime** [comment]
I've started the conversation at the original issue - https://github.com/localstack/localstack/issues/4625#issuecomment-926555052

**Comment by joakime** [comment]
Closing as solved at localstack/localstack#4625
